webapp/javascript/components/FlameGraph/FlameGraphRenderer.jsx

```diff
@@ -57,6 +57,22 @@ class FlameGraphRenderer extends React.Component {
     if (prevState.flamegraphConfigs !== this.state.flamegraphConfigs) {
       this.updateFlamegraphDirtiness();
     }
+
+    if (this.props.isSearchLinked) {
+      if (this.props.linkedSearchQuery !== prevState.highlightQuery) {
+        // eslint-disable-next-line react/no-did-update-set-state
```

Comment:
Any special reason for disabling the linting rule?

Reply:
eslint rule disallows updating state inside componentdidupdate. but, it's actually fine if we're updating the state conditionally. https://stackoverflow.com/a/30529826/3073272

Reply:
Fair enough, can you somehow make it clear on the code? Otherwise the next person reading will (probably) have the same question, and that time they probably won't be scavenging through github issues to understand it.

Reply:
done